### DISCUSSION TITLE
Stopping the progress of a running workflow

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#90] [General]
Hi @ralphhanna,

Are the below use cases achieveable?

**Scenario 1:** 
Say, I have a service task named `DoSomeService` which would be configured in the `appDelegate.ts`. Now, during the processing of that service; an exception or an error occurs (a failure). Therefore, I don't want the workflow to progress from that point of failure. How do I stop the progress of the workflow programatically?

**Scenario 2:** 
Say a scenario 1 occurs. Now someone fixes the problem manually. Now, would we able to retrigger the failed task `DoSomeService` to execute again and progress the workflow normally?

**Scenario 3:** 
Provided the above scenarios can be achieved progromatically especially for service tasks, could we do the same things for other tasks?

This also closely aligns with [a discussion](https://github.com/ralphhanna/bpmn-server/discussions/63) we had previously (we didn't reach any definite conclusion with that yet)

Could you please give some clarifications on this? Thank you!

**Comment by ralphhanna** [comment]
Yes, there are several ways to do this, the simplest way is to use **conditional flow**:

![image](https://github.com/ralphhanna/bpmn-server/assets/11893416/189082db-4033-421a-a247-65d5e3378be3)

The service will have a try/catch and sets the item.data.ErrorCode to either 'Auto','Manual' or 'Abort'

Another option is to use **Error Event** or **Escalate Event**

**Reply by MaheshkumarSundaram** [reply]
Hi @ralphhanna,

Thank you for your example. These error codes would work for all the tasks right not just service tasks?!

**Comment by bpmn-ts** [comment]
Yes, but you have to set the value Sent from my iPhoneOn May 25, 2023, at 3:16 AM, Maheshkumar Sundaram ***@***.***> wrote:﻿
Hi @ralphhanna,
Thank you for your example. These error codes would work for all the tasks right not just service tasks?
